packages/grafana-ui/src/components/TimePicker/TimeRangePicker/TimePickerContent.test.tsx

```diff
@@ -1,59 +1,174 @@
+      const container = renderComponent({ value: absoluteValue, isFullscreen: false, history, showHistory: false });
+      expect(container.queryByText(/recently used absolute ranges/i)).not.toBeInTheDocument();
+      expect(container.queryByText(/2019-12-17 07:48:27 to 2019-12-17 07:49:27/i)).not.toBeInTheDocument();
+      expect(container.queryByText(/2019-10-18 07:50:27 to 2019-10-18 07:51:27/i)).not.toBeInTheDocument();
+    });
+
+    it('renders with relative picker', () => {
+      const container = renderComponent({ value: absoluteValue, isFullscreen: false });
+      expect(container.queryByText(/relative time ranges/i)).toBeInTheDocument();
+      expect(container.queryByText(/other quick ranges/i)).toBeInTheDocument();
+    });
+
+    it('renders without relative picker', () => {
+      const container = renderComponent({ value: absoluteValue, isFullscreen: false, hideQuickRanges: true });
+      expect(container.queryByText(/relative time ranges/i)).not.toBeInTheDocument();
+      expect(container.queryByText(/other quick ranges/i)).not.toBeInTheDocument();
+    });
+
+    it('renders with absolute picker when absolute value and quick ranges are visible', () => {
+      const container = renderComponent({ value: absoluteValue, isFullscreen: false });
+      expect(container.queryByLabelText(/timepicker from field/i)).toBeInTheDocument();
+    });
+
+    it('renders with absolute picker when absolute value and quick ranges are hidden', () => {
+      const container = renderComponent({ value: absoluteValue, isFullscreen: false, hideQuickRanges: true });
+      expect(container.queryByLabelText(/timepicker from field/i)).toBeInTheDocument();
+    });
+
+    it('renders without absolute picker when narrow screen and quick ranges are visible', () => {
+      const container = renderComponent({ value: relativeValue, isFullscreen: false });
+      expect(container.queryByLabelText(/timepicker from field/i)).not.toBeInTheDocument();
+    });
+
+    it('renders with absolute picker when narrow screen and quick ranges are hidden', () => {
+      const container = renderComponent({ value: relativeValue, isFullscreen: false, hideQuickRanges: true });
+      expect(container.queryByLabelText(/timepicker from field/i)).toBeInTheDocument();
+    });
+
+    it('renders without timezone picker', () => {
+      const container = renderComponent({ value: absoluteValue, hideTimeZone: true });
+      expect(container.queryByText(/coordinated universal time/i)).not.toBeInTheDocument();
+    });
   });
 });
 
-function createTimeRange(from: string, to: string): TimeRange {
+function noop(): {} {
+  return {};
+}
+
+function renderComponent({
+  value,
+  isFullscreen = true,
+  showHistory = true,
+  history = [],
+  hideQuickRanges = false,
+  hideTimeZone = false,
+}: {
+  value: TimeRange;
+  isFullscreen?: boolean;
+  showHistory?: boolean;
+  history?: TimeRange[];
+  hideQuickRanges?: boolean;
+  hideTimeZone?: boolean;
+}): RenderResult {
```

Comment:
```suggestion
function renderComponent({
  value,
  isFullscreen = true,
  showHistory = true,
  history = [],
  hideQuickRanges = false,
  hideTimeZone = false,
}: PropsWithScreenSize): RenderResult {
```

We can use `PropsWithScreenSize` here (which needs to be exported from `TimePickerContent`) to avoid redeclaring the types.

Reply:
Fixed. The workaround with `Pick` and `Partial` is needed as using just `Partial` will throw error regarding `value`

Reply:
Also, using would require declaring each mandatory prop everytime.